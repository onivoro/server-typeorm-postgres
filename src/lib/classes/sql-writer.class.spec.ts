import { SqlWriter } from './sql-writer.class';

describe(SqlWriter.name, () => {
    describe(SqlWriter.addColumn.name, () => {
        it('worx', () => {
            expect(SqlWriter.addColumn('blah', {
                isPrimary: true, name: 'id', type: 'character varying'
            })).toMatchSnapshot();
        });
    });

    describe(SqlWriter.createTable.name, () => {
        it('worx', () => {
            expect(SqlWriter.createTable('blah', [
                { isPrimary: true, name: 'id', type: 'character varying' },
                { isPrimary: false, name: 'active', type: 'boolean' },
            ])).toMatchSnapshot();
        });
    });
});
