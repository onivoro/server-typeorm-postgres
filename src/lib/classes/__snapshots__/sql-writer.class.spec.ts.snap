// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`SqlWriter addColumn worx 1`] = `"ALTER TABLE "blah" ADD "id" character varying PRIMARY KEY "`;

exports[`SqlWriter createTable worx 1`] = `
"CREATE TABLE "blah" ("id" character varying PRIMARY KEY ,
"active" boolean NOT NULL );"
`;
